app/src/main/java/ch/epfl/sweng/runpharaa/user/settings/AppCompatPreferenceActivity.java

```diff
@@ -25,14 +29,28 @@ protected void onCreate(Bundle savedInstanceState) {
         getDelegate().installViewFactory();
         getDelegate().onCreate(savedInstanceState);
         super.onCreate(savedInstanceState);
+
     }
 
     @Override
     protected void onPostCreate(Bundle savedInstanceState) {
         super.onPostCreate(savedInstanceState);
+        getDelegate().getSupportActionBar().setHomeButtonEnabled(true);
+        getDelegate().getSupportActionBar().setDisplayHomeAsUpEnabled(true);
```

Comment:
Glad you managed to make it work on all activities, hope it wasn't too ~~much of a pain in the ass~~ hard to find !